### DISCUSSION TITLE
Creating First Admin

### DISCUSSION THREAD

**Discussion by thakurritesh19** [drakkan/sftpgo#1437] [General]
According to readme.md in https://github.com/drakkan/sftpgo, 1 way of creating first admin is by enabling create_default_admin in your configuration file and setting the environment variables SFTPGO_DEFAULT_ADMIN_USERNAME and SFTPGO_DEFAULT_ADMIN_PASSWORD. 

I followed this and despite setting those successfully, I am unable to create first admin. 

The reason I want to do this is to automate creating SFTPGo instance on GCP and then initialize it with some predefined Admin Username and Password.

This is the startup script I use:
```
metadata_startup_script = <<-SCRIPT
    #!/bin/bash
    sudo apt-get update
    sudo add-apt-repository -y ppa:sftpgo/sftpgo
    sudo apt update
    sudo apt-get install -y sftpgo

    echo 'export SFTPGO_DEFAULT_ADMIN_USERNAME="user"' | sudo tee -a /etc/environment
    echo 'export SFTPGO_DEFAULT_ADMIN_PASSWORD="password"' | sudo tee -a /etc/environment

    # Apply changes to the current shell
    source /etc/environment

    sudo -s && jq '.data_provider.create_default_admin = true' /etc/sftpgo/sftpgo.json > temp.json && mv temp.json /etc/sftpgo/sftpgo.json

    # Restart the shell to apply changes
    exec bash

    SCRIPT
```

While everything in my script works, when I access "https://IP:8080/web/admin" first time after installing SFTPGo, it still asks me to create first admin. 

I restarted the service like this: 
`sudo systemctl restart sftpgo`

Service failed to restart.
I checked the logs like this:
`journalctl -u sftpgo`

I see this error: 

`{"level":"error","time":"2023-10-17T00:50:46.006","sender":"service","message":"error initializing data provider: to create the default admin you need to set the env vars \"SFTPGO_DEFAULT_ADMIN_USERNAME\" and \"SFTPGO_DEFAULT_ADMIN_PASSWORD\"""}`

Can anyone let me know if I am missing anything here please ? I am indeed setting these ENV vars. I logged into the instance and checked that required ENV vars are globally declared.

**Comment by fntyler** [comment]
@thakurritesh19 try including the environmental variable `SFTPGO_DATA_PROVIDER__CREATE_DEFAULT_ADMIN` (boolean). This is part of the struct `data_provider` as the following:
> create_default_admin, boolean. Before you can use SFTPGo you need to create an admin account. If you open the admin web UI, a setup screen will guide you in creating the first admin account. You can automatically create the first admin account by enabling this setting and setting the environment variables SFTPGO_DEFAULT_ADMIN_USERNAME and SFTPGO_DEFAULT_ADMIN_PASSWORD. You can also create the first admin by loading initial data. This setting has no effect if an admin account is already found within the data provider. Default false.

YAML example:
```yaml
SFTPGO_DATA_PROVIDER__CREATE_DEFAULT_ADMIN: true
SFTPGO_DEFAULT_ADMIN_USERNAME: admin
SFTPGO_DEFAULT_ADMIN_PASSWORD: admin
```
